src/snyk/common/download/downloader.ts

```diff
@@ -78,25 +76,25 @@ export class Downloader {
     }
   }
 
-  public async downloadLs(
-    lsPath: string,
-    platform: LsSupportedPlatform,
+  public async downloadCli(
+    cliPath: string,
+    platform: CliSupportedPlatform,
     expectedChecksum: string,
   ): Promise<Checksum | null> {
     const hash = new Checksum(expectedChecksum);
 
     return this.window.withProgress(messages.progressTitle, async (progress, token) => {
       const [request, requestToken]: [response: Promise<DownloadAxiosResponse>, cancelToken: CancelTokenSource] =
-        await this.lsApi.downloadBinary(platform);
+        await this.cliApi.downloadBinary(platform);
 
       token.onCancellationRequested(async () => {
         requestToken.cancel();
         this.logger.info(messages.downloadCanceled);
-        await this.deleteFileAtPath(lsPath);
+        await this.deleteFileAtPath(cliPath);
       });
 
       progress.report({ increment: 0 });
-      return await this.doDownload(requestToken, token, lsPath, request, hash, progress);
+      return await this.doDownload(requestToken, token, cliPath, request, hash, progress);
```

Comment:
maybe use a temp file path here and only delete old file when everything is successful, sha256 is correct. Afterwards move the temp file to clipath?

Reply:
e.g. in IntelliJ, we do it like this: https://github.com/snyk/snyk-intellij-plugin/blob/1adcc9ca1fe0f5ec56f8534a45d5c3e5d60ac267/src/main/kotlin/io/snyk/plugin/services/download/CliDownloader.kt#L41

Reply:
https://github.com/snyk/vscode-extension/pull/551/files#r1822131908

Reply:
I left the existing logic for downloading / deleting as is.
Currently it deletes it in the beginning if it exists. But we can change it.